### PR TITLE
dev-util/bitrise bump to 1.23.0

### PR DESCRIPTION
**Pull request by koral--**
dev-util/envman bump to 2.1.0
dev-util/stepman bump to 0.10.4

### CODE REVIEW

File: dev-util/bitrise/bitrise-1.23.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+	mkdir -p "${WORKDIR}/src/${EGO_ON}" || die "Couldn't create project dir in GOPATH"
+	mv "${WORKDIR}/${P}" "${WORKDIR}/src/${EGO_ON}/bitrise" || die "Couldn't move sources to GOPATH"
+}
+
+src_compile() {
+	GOPATH="${WORKDIR}" go build -v -o bin/bitrise || die "Couldn't compile bitrise"
+}
+
+src_test() {
+	pushd "${S}/_tests/integration/" || die "Couldn't find integration tests directory"
```
Comment (mgorny) [inline dev-util/bitrise/bitrise-1.23.0.ebuild]: Why do you need to use absolute paths (`${S}`)? Also `>/dev/null`.

File: dev-util/bitrise/bitrise-1.23.0.ebuild [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+	rm envstore_test.go exit_code_test.go global_flag_test.go json_params_test.go log_filter_test.go output_alias_test.go \
+		step_template_test.go trigger_params_test.go update_test.go timeout_test.go || die "Couldn't remove network-dependent tests"
+	popd || die "Couldn't return to ${S} directory"
+	export PULL_REQUEST_ID=""
```
Comment (mgorny) [inline dev-util/bitrise/bitrise-1.23.0.ebuild]: `local -x`

File: dev-util/envman/envman-2.1.0.ebuild [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit golang-build
+
+EGO_ON="github.com/bitrise-io"
+EGO_PN="${EGO_ON}/${PN}"
+
+DESCRIPTION="Environment variable manager for Bitrise CLI."
```
Comment (mgorny) [inline dev-util/envman/envman-2.1.0.ebuild]: This is not a proper sentence, so no `.` at the end.

File: dev-util/stepman/stepman-0.10.4.ebuild [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit golang-build
+
+EGO_ON="github.com/bitrise-io"
+EGO_PN="${EGO_ON}/${PN}"
+
+DESCRIPTION="Step collection manager for Bitrise CLI."
```
Comment (mgorny) [inline dev-util/stepman/stepman-0.10.4.ebuild]: Likewise.

File: dev-util/stepman/stepman-0.10.4.ebuild [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+	mkdir -p "${WORKDIR}/src/${EGO_ON}" || die "Couldn't create project dir in GOPATH"
+	mv "${WORKDIR}/${P}" "${WORKDIR}/src/${EGO_ON}/stepman" || die "Couldn't move sources to GOPATH"
+}
+
+src_compile() {
+	GOPATH="${WORKDIR}" go build -v -o bin/stepman || die "Couldn't compile stepman"
+}
+
+src_test() {
+	export INTEGRATION_TEST_BINARY_PATH="${S}/bin/stepman"
```
Comment (mgorny) [inline dev-util/stepman/stepman-0.10.4.ebuild]: `local -x`

**Comment by mgorny** [comment]
I'm really sorry I failed to mention this before but please split the commits to one commit per package.

**Comment by mgorny** [comment]
(in dependency requirement order)

File: dev-util/envman/envman-2.1.0.ebuild [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+SRC_URI="https://${EGO_PN}/archive/${PV}.tar.gz -> ${P}.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64"
+
+S="${WORKDIR}/src/${EGO_ON}/${PN}"
+
+src_unpack() {
+	unpack ${A}
```
Comment (mgorny) [inline dev-util/envman/envman-2.1.0.ebuild]: Why not `default` as in bitrise ebuild? Also applies to the second package.

File: dev-util/stepman/stepman-0.10.4.ebuild [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	mkdir -p "${WORKDIR}/src/${EGO_ON}" || die "Couldn't create project dir in GOPATH"
+	mv "${WORKDIR}/${P}" "${WORKDIR}/src/${EGO_ON}/stepman" || die "Couldn't move sources to GOPATH"
+}
+
+src_compile() {
+	GOPATH="${WORKDIR}" go build -v -o bin/stepman || die "Couldn't compile stepman"
+}
+
+src_test() {
+        pushd _tests/integration > /dev/null || die "Couldn't find integration tests directory"
```
Comment (mgorny) [inline dev-util/stepman/stepman-0.10.4.ebuild]: Misident.

File: dev-util/bitrise/bitrise-1.23.0.ebuild [hunk 8]
```diff
@@ -0,0 +1,49 @@
...
+
+src_compile() {
+	GOPATH="${WORKDIR}" go build -v -o bin/bitrise || die "Couldn't compile bitrise"
+}
+
+src_test() {
+	pushd _tests/integration > /dev/null || die "Couldn't find integration tests directory"
+	rm envstore_test.go exit_code_test.go global_flag_test.go json_params_test.go log_filter_test.go output_alias_test.go \
+		step_template_test.go trigger_params_test.go update_test.go timeout_test.go || die "Couldn't remove network-dependent tests"
+	popd || die "Couldn't return to ${S} directory"
```
Comment (mgorny) [inline dev-util/bitrise/bitrise-1.23.0.ebuild]: It's kinda silly you `>/dev/null` pushd but not popd.

Comment (koral--) [reply dev-util/bitrise/bitrise-1.23.0.ebuild]: which version is better?

Comment (mgorny) [reply dev-util/bitrise/bitrise-1.23.0.ebuild]: I'd say `>/dev/null` to both. Paths showing up with no explanation in build are weird.

Comment (koral--) [reply dev-util/bitrise/bitrise-1.23.0.ebuild]: Ok, I'll add it in next update. Thanks.

**Pull request closed without merging**